### PR TITLE
feat: async linting

### PR DESCRIPTION
**Pull request by Kha**
This PR runs all linters for a single command (together) on a separate thread from further elaboration, making a first step towards parallelizing the elaborator.

### CODE REVIEW

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

File: src/Lean/Elab/Frontend.lean [hunk 1]
```diff
@@ -133,6 +133,9 @@ def process (input : String) (env : Environment) (opts : Options) (fileName : Op
   let s ← IO.processCommands inputCtx { : Parser.ModuleParserState } (Command.mkState env {} opts)
   pure (s.commandState.env, s.commandState.messages)
 
+@[extern c inline "lean_io_result_mk_ok(lean_box(0))"]
```
Comment (eric-wieser) [inline src/Lean/Elab/Frontend.lean:136]: Woah, since when has this been a thing?

Comment (Kha) [reply src/Lean/Elab/Frontend.lean]: Forever :) ... and it's gone again

**Comment by Kha** [comment]
!bench

**Comment by Kha** [comment]
!bench

**Pull request merged**